### PR TITLE
Extract examples out of README files

### PR DESCRIPTION
**Pull request by martenbjork**
Quick stab at implementing a solution to https://github.com/Shopify/polaris/issues/5613 .

- Removes code examples from the component Readmes
- Puts them into separate `tsx` files
- Fixes broken example code (adds imports, exports, wrapper functions)
- Adds a json file with descriptions of each example

This is a one-time extraction. After this point, we can make subsequent improvements like running prettier over the examples.

### CODE REVIEW

File: polaris-react/src/components/ButtonGroup/examples/outline-button-group-with-segmented-buttons.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+function Example() {
+  return (
+    <AppProvider i18n={translations}>
+      <ButtonGroup segmented>
+  <Button outline>Bold</Button>
+  <Button outline>Italic</Button>
+  <Button outline>Underline</Button>
+</ButtonGroup>
+    </AppProvider>
+  );
```
Comment (martenbjork) [inline polaris-react/src/components/ButtonGroup/examples/outline-button-group-with-segmented-buttons.tsx:16]: A Prettier run is needed 😊

**Review by alex-page — CHANGES_REQUESTED** [review]
This is great but would break the styleguide. We need to commit to archiving the existing one or think of a backwards compatible solution.

**Comment by martenbjork** [comment]
> This is great but would break the styleguide. We need to commit to archiving the existing one or think of a backwards compatible solution.

Oof, true. We could keep the examples in the Readme for now. That would allow the next.js website to use this new file structure for a time and then, when we're ready to remove the styleguide, we re-run the script, including the part that removes the examples from the readmes.

**Comment by alex-page** [comment]
My thought is...

1. Ship new site with existing README.md example structure
2. Replace old site
3. Fix this up and make best examples ever?

**Comment by martenbjork** [comment]
Yeah, I'm OK with that! 👍

**Comment by sam-b-rose** [comment]
@martenbjork should we go ahead and close this one out?

**Comment by romellogoodman** [comment]
@samrose3 , i think it cool to close in favor of #5940

**Pull request closed without merging**